### ISSUE TITLE
Squeeze fails on Symbolic input

### ISSUE THREAD

**Issue by jacobhinkle** [NVIDIA/Fuser#379]
With dynamic fusions we may not be able to prove that an axis is size-1 before it is used in a `squeeze` operation. Currently, squeeze has a static shape API, but even then it requires the squeezed dimensions to be broadcast, and that check fails with `Symbolic` inputs even if they later are concretized to `Broadcast`. The immediate fix is to accept either `Symbolic` or `Broadcast` inputs, then to check that there are no non-`Broadcast` inputs to squeeze after concretization. We should also make the squeeze API dynamic, i.e. hide the static interface and infer the input shape at concretization just the way we do for `reshape`.

**Comment by naoyam** [comment]
I think it's fine to make squeeze symbolic domains. In fact, we could use squeeze to concretize its symbolic input domains. The underlying assumption is that the original unmodified fusion is the ground truth with no invalid expression. For example, if a fusion has `a = b + c`, we assume `b` and `c` have the same shape. Similarly, I think it makes sense to assume that, if there's `squeeze(a, {true, false})`, `a` is a 2D tensor with the first axis being a broadcast.

We could add an optional pass of validating user-given fusions, not just squeeze inputs but also overall shapes. For now, it should be fine to use `squeeze` to concretize symbolic input domains.

**Comment by jacobhinkle** [comment]
Yeah, so if there is a squeeze of a symbolic ID, it implies that its extent is 1. Propagating this back upstream is similar to determining if a broadcast is resolved (#340). In this case, it could even happen before concretization. As you say, for not I'm just going to accept symbolic and check during concretization that input is `Broadcast` when there's a squeeze.
